src/k2/cpo/service/CPOService.cpp

```diff
@@ -491,4 +503,47 @@ CPOService::handlePersistenceClusterGet(dto::PersistenceClusterGetRequest&& requ
     return RPCResponse(Statuses::S200_OK("persistence cluster found"), std::move(response));
 }
 
+
+seastar::future<std::tuple<Status, dto::MetadataLogStreamRegisterResponse>>
+CPOService::handleMetadataLogStreamRegister(dto::MetadataLogStreamRegisterRequest&& request){
+    K2INFO("Received metadata log stream register request for " << request.name);
+    auto logstream = _metadataStreamLog.find(request.name);
+    if (logstream != _metadataStreamLog.end()) {
+        return RPCResponse(Statuses::S409_Conflict("metadata log already exists"), dto::MetadataLogStreamRegisterResponse());
+    }
+    std::vector<dto::MetadataElement> log;
+    dto::MetadataElement element{.name=std::move(request.plogId), .offset=0};
+    log.push_back(std::move(element));
+    _metadataStreamLog[std::move(request.name)] = std::move(log);
+    K2INFO("Register Success!");
+    return RPCResponse(Statuses::S201_Created("metadata log creates successfully"), dto::MetadataLogStreamRegisterResponse());
+}
+
+seastar::future<std::tuple<Status, dto::MetadataLogStreamUpdateResponse>>
+CPOService::handleMetadataLogStreamUpdate(dto::MetadataLogStreamUpdateRequest&& request){
+    K2INFO("Received metadata log stream update request for " << request.name);
+    auto logstream = _metadataStreamLog.find(request.name);
+    if (logstream == _metadataStreamLog.end()) {
+        return RPCResponse(Statuses::S409_Conflict("metadata log does not exist"), dto::MetadataLogStreamUpdateResponse());
+    }
+
+    logstream->second[-1].offset = request.sealedOffset;
```

Comment:
Why is there a negative index here? If it is supposed to be that way, a comment explaining it would be good.

Reply:
There is a bug before. Fixed